connect-inject/container_init.go

```diff
@@ -336,6 +338,8 @@ export CONSUL_GRPC_ADDR="${HOST_IP}:8502"
 {{- end}}
 {{- if .AuthMethod }}
```

Comment:
I realize I could change this up a bit so that it doesn't duplicate the `consul-k8s connect-init \` command by just wrapping the extra cmd line flags with `if {{ .AuthMethod }}` and that is probably how we'll end up with it in the end. TODO?